### PR TITLE
🌐 Add Japanese translation for Tutorial - Query Parameters and String Validations

### PR DESCRIPTION
**Pull request by SwftAlpc**
This PR translates the tutorial/query-params-str-validations.md to Japanese.
Please review tutorial/query-params-str-validations.md

~~Issue #1571 generates a new directory to support Japanese.
PR #1656 generates a new directory for tutorial.
I will fix dependency after #1571 & #1656  are merged.~~

### CODE REVIEW

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+* `description`
+* `deprecated`
+
+文字列のためのバリデーション:
+
+* `min_length`
+* `max_length`
+* `regex`
+
+この例では、`str`の値をバリデーションを宣言する方法を見てきました。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/query-params-str-validations.md:279]: この例では、`str`の値のバリデーションを宣言する方法を見てきました。

**Comment by SwftAlpc** [comment]
@tokusumi 
Thank you for your review!
I fixed it.

**Review by tokusumi — CHANGES_REQUESTED** [review]
most "hl_lines" is two less than original text.

**Review by tokusumi — APPROVED** [review]

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+
+**FastAPI** ではパラメータの追加情報とバリデーションを宣言することができます。
+
+以下のアプリケーションを例にしてみましょう:
+
+```Python hl_lines="9"
+{!../../../docs_src/query_params_str_validations/tutorial001.py!}
+```
+
+クエリパラメータ`q`は`str`型でデフォルトは`None`のため、オプショナルです。
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:11]:
```suggestion
クエリパラメータ `q` は `Optional[str]` 型で、`None` を許容する `str` 型を意味しており、デフォルトは `None` です。そのため、FastAPIはそれが必須ではないと理解します。
```
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:12]:
```suggestion

!!! note "備考"
    FastAPIは、 `q` はデフォルト値が `=None` であるため、必須ではないと理解します。

    `Optional[str]` における `Optional` はFastAPIには利用されませんが、エディターによるより良いサポートとエラー検出を可能にします。
```

(Missing "note" section translation)

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+```Python hl_lines="9"
+{!../../../docs_src/query_params_str_validations/tutorial002.py!}
+```
+
+デフォルト値`None`を`Query(None)`に置き換える必要があるので、`Query`の最初の引数はデフォルト値を定義するのと同じです。
+
+なので:
+
+```Python
+q: str = Query(None)
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:38]:
```suggestion
q: Optional[str] = Query(None)
```

Original document has `Optional`

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+```
+
+...を以下と同じようにパラメータをオプションにします:
+
+```Python
+q: str = None
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:44]:
```suggestion
q: Optional[str] = None
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+```
+
+しかし、これはクエリパラメータとして明示的に宣言しています。
+
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:48]:
```suggestion

!!! info "情報"
    FastAPIは以下の部分を気にすることを覚えておいてください:

    ```Python
    = None
    ```

    もしくは:

    ```Python
    = Query(None)
    ```

    そして、 `None` を利用することでクエリパラメータが必須ではないと検知します。

    `Optional` の部分は、エディターによるより良いサポートを可能にします。

```

(Missing info section)

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+そして、さらに多くのパラメータを`Query`に渡すことができます。この場合、文字列に適用される、`max_length`パラメータを指定します。
+
+```Python
+q: str = Query(None, max_length=50)
+```
+
+これにより、データを検証し、データが有効でない場合は明確なエラーを表示し、OpenAPIスキーマの*path operation*でパラメータを文書化します。
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:55]:
```suggestion
これにより、データを検証し、データが有効でない場合は明確なエラーを表示し、OpenAPIスキーマの　*path operation* にパラメータを記載します。
```

more readable ?

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+
+この特定の正規表現は受け取ったパラメータの値をチェックします:
+
+* `^`: は、これ以降の文字で始まり、これより以前には文字はありません。
+* `fixedquery`: は、正確な`fixedquery`を持っています.
+* `$`: で終わる場合、`fixedquery`以降には文字はありません.
+
+もしこれらすべての **正規表現**のアイデアについて迷っていても、心配しないでください。多くの人にとって難しい話題です。正規表現を必要としなくても、まだ、多くのことができます。
+
+しかし、あなたがそれらを必要とし、学ぶときにはすでに、 **FastAPI**で直接それらを使用することができることを知っています。
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:81]:
```suggestion
しかし、あなたがそれらを必要とし、学ぶときにはすでに、 **FastAPI**で直接それらを使用することができます。
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 9]
```diff
@@ -0,0 +1,281 @@
...
+第一引数に`None`を渡して、デフォルト値として使用するのと同じように、他の値を渡すこともできます。
+
+クエリパラメータ`q`の`min_length`を`3`とし、デフォルト値を`fixedquery`としてみましょう:
+
+```Python hl_lines="7"
+{!../../../docs_src/query_params_str_validations/tutorial005.py!}
+```
+
+!!! note "備考"
+    デフォルト値を指定すると、パラメータはオプションになります。
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:94]:
```suggestion
    デフォルト値を指定すると、パラメータは任意になります。
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 10]
```diff
@@ -0,0 +1,281 @@
...
+これ以上、バリデーションやメタデータを宣言する必要のない場合は、デフォルト値を指定しないだけでクエリパラメータ`q`を必須にすることができます。以下のように:
+
+```Python
+q: str
+```
+
+以下の代わりに:
+
+```Python
+q: str = None
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:107]:
```suggestion
q: Optional[str] = None
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 11]
```diff
@@ -0,0 +1,281 @@
...
+```
+
+現在は以下の例のように`Query`で宣言しています:
+
+```Python
+q: str = Query(None, min_length=3)
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:113]:
```suggestion
q: Optional[str] = Query(None, min_length=3)
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 12]
```diff
@@ -0,0 +1,281 @@
...
+そのため、`Query`を使用して必須の値を宣言する必要がある場合は、第一引数に`...`を使用することができます:
+
+```Python hl_lines="7"
+{!../../../docs_src/query_params_str_validations/tutorial006.py!}
+```
+
+!!! info "情報"
+    これまで`...`を見たことがない方へ: これは特殊な単一値です。<a href="https://docs.python.org/3/library/constants.html#Ellipsis" class="external-link" target="_blank">Pythonの一部であり、"Ellipsis"と呼ばれています</a>。
+
+これは **FastAPI** にこのパラメータが必要であることを知らせます。
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:125]:
```suggestion
これは **FastAPI** にこのパラメータが必須であることを知らせます。
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 13]
```diff
@@ -0,0 +1,281 @@
...
+
+## クエリパラメータのリスト/複数の値
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:127]:
```suggestion
## クエリパラメータのリスト / 複数の値
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 14]
```diff
@@ -0,0 +1,281 @@
...
+```
+
+!!! tip "豆知識"
+    上述の例のように、`list`型のクエリパラメータを宣言するには明示的に`Query`を使用する必要があります。そうしない場合、リクエストボディと解釈されます。
+
+対話的APIドキュメントは複数の値を許可するために自動的に更新されます。
+
+<img src="https://fastapi.tiangolo.com/img/tutorial/query-params-str-validations/image02.png">
+
+### デフォルト値を持つ、クエリパラメータのリスト/複数の値
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:163]:
```suggestion
### デフォルト値を持つ、クエリパラメータのリスト / 複数の値
```

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 15]
```diff
@@ -0,0 +1,281 @@
...
+```Python hl_lines="7"
+{!../../../docs_src/query_params_str_validations/tutorial013.py!}
+```
+
+!!! note "備考"
+    この場合、FastAPIはリストの内容をチェックしないことを覚えておいてください。
+
+    例えば`List[int]`はリストの内容が整数であるかどうかをチェックします(そして、文書化します)。しかし`list`だけではそうしません。
+
+## さらにメタデータを宣言する
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:201]:
```suggestion
## より多くのメタデータを宣言する
```
more fluent ?

File: docs/ja/docs/tutorial/query-params-str-validations.md [hunk 16]
```diff
@@ -0,0 +1,281 @@
...
+http://127.0.0.1:8000/items/?item-query=foobaritems
+```
+
+しかし、`item-query`は有効なPythonの変数名ではありません。
+
+最も近いのは`item_query`でしょう。
+
+しかし、どうしても`item-query`と正確に一致している必要があるとします...
+
+その時、`alias`を宣言することができます。エイリアスはパラメータの値を見つけるのに使用されます:
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/query-params-str-validations.md:240]:
```suggestion
それならば、`alias`を宣言することができます。エイリアスはパラメータの値を見つけるのに使用されます:
```

**Review by Attsun1031 — COMMENTED** [review]
Thank you for your great work !
Please check suggestions I made.

**Comment by SwftAlpc** [comment]
@Attsun1031 
Thanks for the review!
Could you re-review it that has been corrected?

**Review by Attsun1031 — APPROVED** [review]
LGTM 👍

**Comment by tiangolo** [comment]
Awesome, thanks @SwftAlpc ! :clap: 

And thanks @Attsun1031  and @tokusumi for the reviews :bow:

**Pull request merged**